### PR TITLE
Issue 2252 pdb missing cryst

### LINKED ISSUE #2252: MDAnalysis unable to read a pdb with mangled CRYST record
6edu.pdb cannot be read by MDAnalysis (here mda)

script:
```
u = mda.Universe('6edu.pdb')
```
Error:
```
Traceback (most recent call last):
  File "test_mda.py", line 25, in <module>
    u = mda.Universe('6edu.pdb',format="xpdb")
  File "/Users/trovato/anaconda2/envs/pdb_analysis/lib/python3.6/site-packages/MDAnalysis/core/universe.py", line 335, in __init__
    self.load_new(coordinatefile, **kwargs)
  File "/Users/trovato/anaconda2/envs/pdb_analysis/lib/python3.6/site-packages/MDAnalysis/core/universe.py", line 602, in load_new
    self.trajectory = reader(filename, **kwargs)
  File "/Users/trovato/anaconda2/envs/pdb_analysis/lib/python3.6/site-packages/MDAnalysis/coordinates/PDB.py", line 323, in __init__
    self._read_frame(0)
  File "/Users/trovato/anaconda2/envs/pdb_analysis/lib/python3.6/site-packages/MDAnalysis/coordinates/PDB.py", line 391, in _read_frame
    line[40:47], line[47:54]]
ValueError: could not convert string to float: 
```
Can the coordinates of the order of 100 Ang be a problem? How shall I read in this pdb?

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #2252

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/coordinates/PDB.py [hunk 1]
```diff
@@ -389,9 +389,13 @@ def _read_frame(self, frame):
...
                 # does an implicit str -> float conversion
-                self.ts._unitcell[:] = [line[6:15], line[15:24],
-                                        line[24:33], line[33:40],
-                                        line[40:47], line[47:54]]
+                try:
+                    self.ts._unitcell[:] = [line[6:15], line[15:24],
+                                            line[24:33], line[33:40],
+                                            line[40:47], line[47:54]]
+                except ValueError:
+                    warnings.warn("Failed to read CRYST entry, got:\n{}"
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/PDB.py]: Maybe be even more explicit and say "Failed to read CRYST1 record, possibly invalid PDB file: read:\n{}"

Comment (orbeckst) [reply package/MDAnalysis/coordinates/PDB.py]: Just to make sure that user's understand that this is a problem with the file and not with MDAnalysis.

**Comment by richardjgowers** [comment]
@orbeckst thanks for review.  To clarify, these files will still be read, but the cryst records will be ignored (with warnings).

**Comment by orbeckst** [comment]
Oh, yes – should have read the code properly.

What do you think about actually raising an error? Warnings are easily overlooked. A missing unit cell might lead to various errors downstream.

**Comment by richardjgowers** [comment]
@orbeckst the file are straight from the PDB, so while they're not correct, they're also not really incorrect.  I think a user can expect to load something they get from that source.  Functions further down the line (that deal with PBC) should be throwing errors when given a 0 sized box

**Comment by orbeckst** [comment]
Fair enough. Then merge, please. 

--
Oliver Beckstein
email: orbeckst@gmail.com

**Pull request merged**